nsv13/code/modules/overmap/weapons/plasma_gun.dm

```diff
@@ -0,0 +1,466 @@
+		fool.IgniteMob()
+		playsound(usr.loc, 'sound/magic/lightningbolt.ogg', 100, 1, extrarange = 30)
+		return
+	else
+		visible_message("<span class 'notice'>[usr] wisely lowers their crowbar.")
+		return
+
+/obj/machinery/ship_weapon/plasma_caster/multitool_act(mob/living/user, obj/item/I)
+	. = TRUE
+	if(maint_state == 0)
+		to_chat(user, "<span class='notice'>You must first open the maintenance panel before unwrenching the protective casing!</span>")
+	if(maint_state == 1)
+		to_chat(user, "<span class='notice'>You must unbolt the protective casing before tuning the magnetic frequency!</span>")
+	else
+		to_chat(user, "<span class='notice'>You being tuning the magnetic frequency.</span>")
+		while(alignment < 100)
+			if(!do_after(user, 5, target = src))
+				return
+			alignment += rand(1,2)
+			if(alignment >= 100)
+				alignment = 100
+				break
+
+/obj/machinery/ship_weapon/plasma_caster/attack_hand(mob/user)
+	ui_interact(user)
+	return
+
+/obj/machinery/ship_weapon/plasma_caster/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "PlasmaGun")
+		ui.open()
+		ui.set_autoupdate(TRUE)
+
+/obj/machinery/ship_weapon/plasma_caster/ui_data(mob/user)
+	. = ..()
+	var/list/data = list()
+	data["field_integrity"] = field_integrity
+	data["alignment"] = alignment
+	data["plasma_moles"] = plasma_mole_amount
+	data["plasma_moles_max"] = plasma_fire_moles
+	data["safety"] = safety
+	data["loaded"] = (state > STATE_LOADED) ? TRUE : FALSE
+	data["chambered"] = (state > STATE_FED) ? TRUE : FALSE
+	data["ammo"] = ammo?.len || 0
+	data["max_ammo"] = max_ammo
+	data["cooldown"] = cooldown
+	return data
+
+/obj/machinery/ship_weapon/plasma_caster/ui_act(action, params)
+	if(..())
+		return
+	switch(action)
+		if("toggle_load")
+			if(state == STATE_LOADED)
+				feed()
+			else
+				unload()
+		if("chamber")
+			chamber()
+
+		if("toggle_safety")
+			toggle_safety()
+
+	return
+
+/datum/ship_weapon/plasma_caster
+	name = "MPAC"
+	burst_size = 1
+	fire_delay = 5 SECONDS //Everyone's right, weapon code is jank...
+	range = 25000 //It will continue to
+	default_projectile_type = /obj/item/projectile/bullet/plasma_caster
+	select_alert = "<span class='notice'>Charging magnetic accelerator...</span>"
+	failure_alert = "<span class='warning'>Magnetic Accelerator not ready!</span>"
+	overmap_firing_sounds = list('nsv13/sound/effects/ship/plasma_gun_fire.ogg')
+	overmap_select_sound = 'nsv13/sound/effects/ship/phaser_select.ogg'
+	weapon_class = WEAPON_CLASS_HEAVY
+	ai_fire_delay = 180 SECONDS
+	allowed_roles = OVERMAP_USER_ROLE_GUNNER
+	lateral = FALSE
+
+/obj/item/ship_weapon/ammunition/plasma_core
+	name = "\improper Condensed Phoron Core"
+	desc = "A heavy, condensed ball of phoron coated in a thick shell to prevent accidents."
+	icon = 'nsv13/icons/obj/munitions.dmi'
+	icon_state = "plasma_core"
+	projectile_type = /obj/item/projectile/bullet/plasma_caster
```

Comment:
You should totally make a gibtonite version of this...

Reply:
I'd probably do it with an update PR later or something.